data/crac-creation/crac-creator-csa-profiles/src/main/java/com/farao_community/farao/data/crac_creation/creator/csa_profile/crac_creator/remedial_action/CsaProfileRemedialActionsCreator.java

```diff
@@ -0,0 +1,253 @@
+
+        for (PropertyBag parentRemedialActionPropertyBag : gridStateAlterationRemedialActionPropertyBags) {
+            String remedialActionId = parentRemedialActionPropertyBag.get(CsaProfileConstants.MRID);
+
+            if (!linkedTopologyActions.containsKey(remedialActionId)) {
+                continue;
+            }
+
+            if (importRemedialAction(parentRemedialActionPropertyBag)) {
+                Optional<String> nativeRaNameOpt = Optional.ofNullable(parentRemedialActionPropertyBag.get(CsaProfileConstants.REMEDIAL_ACTION_NAME));
+                Optional<String> tsoNameOpt = Optional.ofNullable(parentRemedialActionPropertyBag.get(CsaProfileConstants.TSO));
+                Optional<String> targetRemedialActionNameOpt = createRemedialActionName(nativeRaNameOpt.orElse(null), tsoNameOpt.orElse(null));
+                Optional<Integer> speedOpt = getSpeedOpt(parentRemedialActionPropertyBag.get(CsaProfileConstants.TIME_TO_IMPLEMENT));
+
+                NetworkActionAdder networkActionAdder = crac.newNetworkAction();
+                networkActionAdder.withId(remedialActionId);
+                targetRemedialActionNameOpt.ifPresent(networkActionAdder::withName);
+                tsoNameOpt.ifPresent(tso -> networkActionAdder.withOperator(tso.substring(33)));
+                speedOpt.ifPresent(networkActionAdder::withSpeed);
+
+                if (linkedContingencyWithRAs.containsKey(remedialActionId)) {
+                    String combinationConstraintKind = linkedContingencyWithRAs.get(remedialActionId).iterator().next().get(CsaProfileConstants.COMBINATION_CONSTRAINT_KIND); // any kind
+                    if (!checkAllContingenciesLinkedToRaHaveTheSameConstraintKind(remedialActionId, linkedContingencyWithRAs.get(remedialActionId), combinationConstraintKind)) {
+                        return;
+                    }
+
+                    List<String> faraoContingenciesIds = new ArrayList<>();
+                    for (PropertyBag contingencyWithRemedialActionPropertyBag : linkedContingencyWithRAs.get(remedialActionId)) {
+                        importOnStateRa(faraoContingenciesIds, contingencyWithRemedialActionPropertyBag, parentRemedialActionPropertyBag, remedialActionId, combinationConstraintKind);
+                    }
+
+                    if (!faraoContingenciesIds.isEmpty()) {
+                        fillContingencies(networkActionAdder, faraoContingenciesIds, linkedTopologyActions.get(remedialActionId), combinationConstraintKind);
+                    } else {
+                        csaProfileRemedialActionCreationContexts.add(CsaProfileRemedialActionCreationContext.notImported(remedialActionId, ImportStatus.INCONSISTENCY_IN_DATA, "None of the remedial actions with contingency linked to the grid state alteration with id: " + remedialActionId + " matches a contingency that has has imported"));
+                    }
```

Comment:
but from what I understand reading you code, you want the whole RA import to fail if at least one contingency is wrong